src/script/descriptor.cpp

```diff
@@ -246,18 +246,19 @@ enum class DeriveType {
 /** An object representing a parsed extended public key in a descriptor. */
 class BIP32PubkeyProvider final : public PubkeyProvider
 {
-    CExtPubKey m_extkey;
+    // Root xpub, path, and final derivation step type being used, if any
+    CExtPubKey m_root_extkey;
```

Comment:
It's not necessarily the root, right? For example in `[012345678/0']xpub.../0/1` this variable would correspond to the 2nd level intermediary node.

Reply:
Sure, but it's easier to just call this the root and it's the root key that we do all our later derivation from.

Reply:
I think there's enough bikeshedding in this pr :/